components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java

```diff
@@ -655,6 +668,74 @@ private static String getComponentCategory(String ref) {
         return "";
     }
 
+    /**
+     * Extract the archive file and validates the openAPI definition
+     *
+     * @param inputStream   file as input stream
+     * @param returnContent whether to return the content of the definition in the response DTO
+     * @return APIDefinitionValidationResponse
+     * @throws APIManagementException if error occurred while parsing definition
+     */
+    public static APIDefinitionValidationResponse extractAndValidateOpenAPIArchive(InputStream inputStream,
+            boolean returnContent) throws APIManagementException {
+        String path = System.getProperty(APIConstants.JAVA_IO_TMPDIR) + File.separator +
+                APIConstants.OPENAPI_ARCHIVES_TEMP_FOLDER + File.separator + UUID.randomUUID().toString();
+        String archivePath = path + File.separator + APIConstants.OPENAPI_ARCHIVE_ZIP_FILE;
+        String extractedLocation = APIFileUtil
+                .extractUploadedArchive(inputStream, APIConstants.OPENAPI_EXTRACTED_DIRECTORY, archivePath, path);
+        File[] listOfFiles = new File(extractedLocation).listFiles();
+        File archiveDirectory = null;
+        if (listOfFiles != null) {
+            for (File file: listOfFiles) {
+                if (file.isDirectory()) {
+                    archiveDirectory = file.getAbsoluteFile();
```

Comment:
you can exit the loop once you identify the archiveDirectory.

Reply:
This check is to get the directory path inside the zip and avoid getting files instead of directories. Thus the zip always contain only one directory.
Valid input :
`temp.zip --- > temp (directory)`
Invalid input :
`temp.zip ----> temp.yaml`

Reply:
we agreed to use the naming convention for the master swagger. Because we cannot guarantee the directory structure because helper files may contain multiple directories in same level. 
So we can remove this logic.